app/src/main/java/com/larryhsiao/nyx/android/tag/TagListFragment.java

```diff
@@ -0,0 +1,178 @@
+                    }
+                }).show();
+            return new Object();
+        });
+        final RecyclerView list = view.findViewById(R.id.list);
+        list.setLayoutManager(new LinearLayoutManager(view.getContext()));
+        list.setAdapter(adapter);
+        adapter.loadTags(
+            new QueriedTags(
+                new AllTags(db)
+            ).value().stream().sorted(new Comparator<Tag>() {
+                final StringComparator comparator = new StringComparator();
+
+                @Override
+                public int compare(Tag o1, Tag o2) {
+                    return comparator.compare(o2.title(), o1.title());
+                }
+            }).collect(Collectors.toList())
+        );
+    }
+
+    @Override
+    public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+        inflater.inflate(R.menu.jot_list, menu);
+
+        SearchManager searchManager = ((SearchManager) getContext().getSystemService(Context.SEARCH_SERVICE));
+        MenuItem searchMenuItem = menu.findItem(R.id.menuItem_search);
+        SearchView searchView = (SearchView) searchMenuItem.getActionView();
+        searchView.setSearchableInfo(searchManager.getSearchableInfo(getActivity().getComponentName()));
+        searchView.setOnCloseListener(() -> {
+            searchMenuItem.collapseActionView();
+            return false;
+        });
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                return false;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String newText) {
+                StringComparator comparator = new StringComparator();
+                adapter.loadTags(
+                    new QueriedTags(new TagsByKeyword(db, newText))
+                        .value()
+                        .stream()
+                        .sorted((o1, o2) ->
+                            comparator.compare(o2.title(), o1.title())
+                        ).collect(Collectors.toList())
+                );
+                return true;
+            }
+        });
+        searchMenuItem.setOnActionExpandListener(new MenuItem.OnActionExpandListener() {
+            @Override
+            public boolean onMenuItemActionExpand(MenuItem item) {
+                return true;
+            }
+
+            @Override
+            public boolean onMenuItemActionCollapse(MenuItem item) {
+                searchView.setQuery("", true);
+                return true;
+            }
+        });
+        searchMenuItem.setOnMenuItemClickListener(item -> {
+            searchView.onActionViewExpanded();
+            return false;
+        });
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+        switch (item.getItemId()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Switch statements should have a default label](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5112315)